sdk/storage/azure-storage-blob/azure/storage/blob/_container_client.py

```diff
@@ -64,7 +65,7 @@ def _get_blob_name(blob):
     :rtype: str
     """
     try:
-        return blob.name
+        return blob.get('name')
```

Comment:
This is a significant change. Why was this made?

Reply:
because when we send a batch request and the blob type is a dict, this get_name method doesn't work. blob is either a BlobProperties or a str, while it's possible that users provided a customized class which has name attribute.
How about do this:
`try:
    return blob.name
except AttributeError:
    try:
        return blob.get('name'):
   except:
        return blob

`